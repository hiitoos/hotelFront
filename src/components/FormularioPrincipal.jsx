import React, { useState } from 'react';
import DatePicker, { registerLocale } from "react-datepicker";
import es from 'date-fns/locale/es';
import "react-datepicker/dist/react-datepicker.css";
registerLocale('es', es);

const initialFilter = {
    price_from: '',
    price_to: '',
    type: '',
    guest_from: '',
    date_from: new Date(),
    date_to: new Date(),
}

function Search(props) {

    const [filter, setFilter] = useState(initialFilter);
    console.log(props);

    function handleChange(event) {
        setFilter({
            ...filter,
            [event.target.name]: event.target.value,
        });
        props.onFilterChange({
            ...filter,
            [event.target.name]: event.target.value,
        })
    }

    const handleChangeDateIn = (date) => {
        setFilter({
            ...filter,
            date_from: date,
        });
        props.onFilterChange({
            ...filter,
            date_from: date,
        })
    }

    const handleChangeDateOut = (date) => {
        setFilter({
            ...filter,
            date_to: date,
        });
        props.onFilterChange({
            ...filter,
            date_to: date,
        })
    }

    return (
        <div className='search'>
            <div className='filter'>
                <input type='number' placeholder='price from' name='price_from' value={filter.price_from}
                    onChange={handleChange} />
                <input type='number' placeholder='price to' name='price_to' value={filter.price_to}
                    onChange={handleChange} />
                <input type='number' placeholder='guest_from' name='guest_from' value={filter.guest_from}
                    onChange={handleChange} />
                <input type='text' placeholder='type' name='type' value={filter.type} onChange={handleChange} />
                <DatePicker 
                    name='date_from' 
                    dateFormat="dd/MM/yyyy"
                    value={filter.date_from==='' ? new Date() : filter.date_from} 
                    onChange={handleChangeDateIn} 
                    selected={filter.date_from}
                    minDate={new Date()}
                    locale="es"
                />
                <DatePicker 
                    name='date_to' 
                    dateFormat="dd/MM/yyyy"
                    value={filter.date_to} 
                    onChange={handleChangeDateOut} 
                    selected={filter.date_to<filter.date_from?filter.date_from:filter.date_to}
                    minDate={filter.date_from}
                    locale="es"
                />
            </div>
        </div>
  );
}

export default Search;
