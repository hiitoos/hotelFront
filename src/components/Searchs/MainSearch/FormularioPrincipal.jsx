import React, { useState } from 'react';
import {
    Container,
    Row,
    Col
} from "reactstrap";
import DatePicker, { registerLocale } from "react-datepicker";
import es from 'date-fns/locale/es';
import "react-datepicker/dist/react-datepicker.css";
import { now } from '../../../services/date.service';
registerLocale('es', es);

const initialFilter = {
    price_from: '',
    price_to: '',
    type: '',
    date_from: now,
    date_to: now,
}

function Search(props) {

    const [filter, setFilter] = useState(initialFilter);

    function handleChange(event) {
        setFilter({
            ...filter,
            [event.target.name]: event.target.value,
        });
        props.onFilterChange({
            ...filter,
            [event.target.name]: event.target.value,
        })
    }

    const handleChangeDateIn = (date) => {
        setFilter({
            ...filter,
            date_from: date,
        });
        props.onFilterChange({
            ...filter,
            date_from: date,
        })
    }

    const handleChangeDateOut = (date) => {
        setFilter({
            ...filter,
            date_to: date,
        });
        props.onFilterChange({
            ...filter,
            date_to: date,
        })
    }

    return (
        <div className='search'>
            <Container className='filter'>
                <Row>
                    <Col lg={12}>
                        <label to='price_from'>A partir de ... €</label><br/>
                        <input type='number' placeholder='-------------------------------' name='price_from' min={0} value={filter.price_from}
                        onChange={handleChange} />
                    </Col>
                </Row>
                <Row>
                    <Col lg={12}>
                        <label to='price_to'>Hasta ... €</label><br/>
                        <input type='number' placeholder='-------------------------------' name='price_to' min={0} value={filter.price_to}
                        onChange={handleChange} />
                    </Col>
                </Row>
                <Row>
                    <Col lg={12}>
                        <label to='type'>Tipo de habitación</label><br/>
                        <input type='text' placeholder='Simple, Doble, Enorme...' name='type' value={filter.type} onChange={handleChange} />
                    </Col>
                </Row>
                <Row className="datepickers">
                    <Col lg={12}>
                        <label to='date_from'>Fecha de entrada</label><br/>
                        <DatePicker
                            className='datepicker_from' 
                            name='date_from' 
                            dateFormat="dd/MM/yyyy"
                            value={filter.date_from} 
                            onChange={handleChangeDateIn} 
                            selected={filter.date_from}
                            minDate={now}
                            locale="es"
                        />
                    </Col>
                </Row>
                <Row className="datepickers">
                    <Col lg={12}>
                    <label to='date_to'>Fecha de salida</label><br/>
                        <DatePicker 
                            className='datepicker_to' 
                            name='date_to' 
                            dateFormat="dd/MM/yyyy"
                            value={filter.date_to} 
                            onChange={handleChangeDateOut} 
                            selected={filter.date_to<filter.date_from?filter.date_from:filter.date_to}
                            minDate={filter.date_from}
                            locale="es"
                        />
                    </Col>
                </Row>
            </Container>
        </div>
  );
}

export default Search;
