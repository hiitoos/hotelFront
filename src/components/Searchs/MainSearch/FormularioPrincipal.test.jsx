import React from 'react'
import FormularioPrincipal from './FormularioPrincipal.jsx'
import DatePicker from 'react-datepicker'
import { shallow, mount } from 'enzyme'
import { now } from '../../../services/date.service'

describe ('MainSearch unit test', () => {
    test('it should render without crashing', () =>{
        const wrapper = shallow (<FormularioPrincipal />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should have required inputs', () => {
        const wrapper = shallow (<FormularioPrincipal />);
        expect(wrapper.find('input[name="price_from"]')).toHaveLength(1);
        expect(wrapper.find('input[name="price_to"]')).toHaveLength(1);
        expect(wrapper.find('input[name="type"]')).toHaveLength(1);
    })

    test('should run onChange PRICE FROM with the new State', () => {
        const mockUpdateFilter = jest.fn();
        const wrapper = shallow (<FormularioPrincipal onFilterChange={mockUpdateFilter}/>);
        wrapper.find('input[name="price_from"]').simulate('change', {target:{name:'price_from', value: '100'}});
        expect (mockUpdateFilter).toHaveBeenCalledWith({
            price_from: '100',
            price_to: '',
            type: '',
            date_from: now,
            date_to: now,
        })
    })

    test('should run onChange PRICE TO with the new State', () => {
        const mockUpdateFilter = jest.fn();
        const wrapper = shallow (<FormularioPrincipal onFilterChange={mockUpdateFilter}/>);
        wrapper.find('input[name="price_to"]').simulate('change', {target:{name:'price_to', value: '200'}});
        expect (mockUpdateFilter).toHaveBeenCalledWith({
            price_from: '',
            price_to: '200',
            type: '',
            date_from: now,
            date_to: now,
        })
    })

    test('should run onChange TYPE with the new State', () => {
        const mockUpdateFilter = jest.fn();
        const wrapper = shallow (<FormularioPrincipal onFilterChange={mockUpdateFilter}/>);
        wrapper.find('input[name="type"]').simulate('change', {target:{name:'type', value: 'Simple'}});
        expect (mockUpdateFilter).toHaveBeenCalledWith({
            price_from: '',
            price_to: '',
            type: 'Simple',
            date_from: now,
            date_to: now,
        })
    })

})