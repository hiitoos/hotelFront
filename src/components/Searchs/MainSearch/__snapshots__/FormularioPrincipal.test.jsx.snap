// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSearch unit test it should render without crashing 1`] = `ShallowWrapper {}`;
